File: app/src/main/java/de/blankedv/sx4control/Preferences.java
package de.blankedv.sx4control;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
import android.os.Bundle;
import android.preference.EditTextPreference;
import android.preference.ListPreference;
import android.preference.PreferenceActivity;
import android.preference.PreferenceManager;

import static de.blankedv.sx4control.Constants.DEBUG;
import static de.blankedv.sx4control.Constants.KEY_IP;


public class Preferences extends PreferenceActivity implements OnSharedPreferenceChangeListener {


    private EditTextPreference ipPref, portPref;
    //private ListPreference configFilenamePref, locosFilenamePref;


    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        addPreferencesFromResource(R.xml.preferences);

        ipPref = (EditTextPreference) getPreferenceScreen().findPreference(KEY_IP);
        //relFontSize = (ListPreference) getPreferenceScreen().findPreference(KEY_FONT_FACTOR);

    }

    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences,
                                          String key) {

        // Let's do something if a preference value changes
        if (key.equals(KEY_IP)) {
            ipPref.setSummary("= " + sharedPreferences.getString(KEY_IP, ""));
        }

    }

    @Override
    protected void onResume() {
        super.onResume();

        Context context = getApplicationContext();
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        // Setup the initial values
        ipPref.setSummary("= " + prefs.getString(KEY_IP, ""));

        // Set up a listener whenever a key changes
        getPreferenceScreen().getSharedPreferences().registerOnSharedPreferenceChangeListener(this);
    }

    @Override
    protected void onPause() {
        super.onPause();

        // Unregister the listener whenever a key changes
        getPreferenceScreen().getSharedPreferences().unregisterOnSharedPreferenceChangeListener(this);
    }





}

File: app/src/main/java/de/blankedv/sx4control/Preferences.kt
package de.blankedv.sx4control

import android.content.Context
import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.os.Bundle
import android.preference.EditTextPreference
import android.preference.ListPreference
import android.preference.PreferenceActivity
import android.preference.PreferenceManager

class Preferences : PreferenceActivity(), OnSharedPreferenceChangeListener {


    private var ipPref: EditTextPreference? = null
    private val portPref: EditTextPreference? = null
    //private ListPreference configFilenamePref, locosFilenamePref;


    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        addPreferencesFromResource(R.xml.preferences)

        ipPref = preferenceScreen.findPreference(KEY_IP) as EditTextPreference
        //relFontSize = (ListPreference) getPreferenceScreen().findPreference(KEY_FONT_FACTOR);

    }

    override fun onSharedPreferenceChanged(
        sharedPreferences: SharedPreferences,
        key: String
    ) {

        // Let's do something if a preference value changes
        if (key == KEY_IP) {
            ipPref!!.summary = "= " + sharedPreferences.getString(KEY_IP, "")!!
        }

    }

    override fun onResume() {
        super.onResume()

        val context = applicationContext
        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
        // Setup the initial values
        ipPref!!.summary = "= " + prefs.getString(KEY_IP, "")!!

        // Set up a listener whenever a key changes
        preferenceScreen.sharedPreferences.registerOnSharedPreferenceChangeListener(this)
    }

    override fun onPause() {
        super.onPause()

        // Unregister the listener whenever a key changes
        preferenceScreen.sharedPreferences.unregisterOnSharedPreferenceChangeListener(this)
    }


}

